embedded/store/kv_metadata.go

```diff
@@ -15,43 +15,200 @@ limitations under the License.
 */
 package store
 
-const maxKVMetadataLen = 1
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"time"
+)
+
+var ErrNonExpirable = errors.New("non expirable")
+
+const (
+	deletedAttrCode   attributeCode = iota
+	expiresAtAttrCode attributeCode = iota
```

Comment:
Post-merge: Since those values are used in on-disk format, I'd use concrete values here:

```suggestion
	deletedAttrCode   attributeCode = 0
	expiresAtAttrCode attributeCode = 1
```

otherwise it would be too easy to break things by adding new entry in the middle for example